File: frontend/src/types/query.ts
export type QueryResponse = {
    id: number;
    question: string;
    generated_sql: string;
    explanation: string;
    status: string;
    columns: string[];
    rows: unknown[][];
    row_count: number;
    execution_time_ms: number;
    created_at: string;
};

export type HistoryItem = {
    id: number;
    question: string;
    generated_sql: string;
    explanation: string;
    status: string;
    row_count: number | null;
    execution_time_ms: number | null;
    created_at: string;
};

export type ExampleItem = {
    id: number;
    question: string;
};

File: backend/app/api/schemas/query.py
from datetime import datetime

from pydantic import BaseModel, Field


class QueryRequest(BaseModel):
    question: str = Field(..., min_length=1, description="Natural language business question")


class QueryResponse(BaseModel):
    id: int
    question: str
    generated_sql: str
    explanation: str
    status: str
    created_at: datetime